directory.go

```diff
@@ -0,0 +1,43 @@
+//  Copyright (c) 2021 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+)
+
+type Directory interface {
+	GetWriter(filePath string) (io.WriteCloser, error)
+}
+
+// FileSystemDirectory is the default implementation for the
+// Directory interface.
+type FileSystemDirectory string
+
+func (f FileSystemDirectory) GetWriter(filePath string) (io.WriteCloser,
+	error) {
+	dir, file := filepath.Split(filePath)
+	if dir != "" {
```

Comment:
So, I had changed this logic to allow for creating any sub-directory needed inside the directory.  But you have changed this back to always/only create the store directory, can you explain why?

Reply:
It's a mistake. I just followed the version you shared here - https://play.golang.org/p/5TkAcff-JM0
```
if dir != "" {
		err := os.MkdirAll(filepath.Join(string(f), dir), os.ModePerm)
		if err != nil {
			return nil, err
		}
	}
```
should be like this.

let me fix it now.